util/src/main/java/io/kubernetes/client/Copy.java

```diff
@@ -121,8 +122,7 @@ public void copyDirectoryFromPod(
             false,
             false);
     try (InputStream is = new Base64InputStream(new BufferedInputStream(proc.getInputStream()));
-        ArchiveInputStream archive = new TarArchiveInputStream(is)) {
-      // TODO Use new GzipCompressorInputStream(is))) here {
+        ArchiveInputStream archive = new TarArchiveInputStream(new GzipCompressorInputStream(is))) {
```

Comment:
i think we need to enable the gzip compression on the remote process too, by applying "z" option? additionally, we need a test for the change.

Reply:
Yep, we definitely need it on both sides, the current encoding is just 'tar'

Reply:
I think
final Process proc =
        this.exec(
            namespace,
            pod,
            new String[] {"sh", "-c", "tar cf**z** - " + srcPath + " | base64"},
            container,
            false,
            false);
Am I thinking right?

Reply:
yes, but you dropped "f", is that intended?

Reply:
Yes- we don't need "f". As we are directly getting the stream from process' stdout.

Reply:
iiuc, if you drop "f", you will also have to drop the tailing "-" from the command. in the previous command `tar cf - ...` where "-" is the argument for "f" from the bundled options. i suggest keep the "f" for clarification given that it's a more commonly used form.